### PR TITLE
add reconcile for suite runner role and role binding

### PR DESCRIPTION
**Pull request by andmat900**
### Applicable Issues
- https://github.com/eiffel-community/etos/issues/300

### Description of the Change
This change adds reconciliation of Suite Runner service account, role and rolebinding.

### Sign-off
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Andrei Matveyeu, andrei.matveyeu@axis.com

### CODE REVIEW

File: internal/etos/suitestarter/suitestarter.go [hunk 1]
```diff
@@ -320,7 +345,7 @@ func (r *ETOSSuiteStarterDeployment) reconcileRolebinding(ctx context.Context, l
 		if !apierrors.IsNotFound(err) {
 			return rolebinding, err
 		}
-		logger.Info("Creating a new role binding for the suite starter")
+		logger.Info(fmt.Sprintf("Creating a new role binding for the suite starter: %s", roleName))
```
Comment (t-persson) [inline internal/etos/suitestarter/suitestarter.go:348]:
```suggestion
		logger.Infof("Creating a new role binding for the suite starter: %s", roleName)
```

File: internal/etos/suitestarter/suitestarter.go [hunk 2]
```diff
@@ -440,6 +465,27 @@ func (r *ETOSSuiteStarterDeployment) role(name types.NamespacedName, labelName s
 	}
 }
 
+// testReader creates a role resource definition giving read permissions for testrun resource
+func (r *ETOSSuiteStarterDeployment) testReaderRole(name types.NamespacedName, labelName string) *rbacv1.Role {
```
Comment (t-persson) [inline internal/etos/suitestarter/suitestarter.go:469]:
```suggestion
func (r *ETOSSuiteStarterDeployment) testRunReaderRole(name types.NamespacedName, labelName string) *rbacv1.Role {
```

File: internal/etos/suitestarter/suitestarter.go [hunk 3]
```diff
@@ -87,53 +92,73 @@ func (r *ETOSSuiteStarterDeployment) Reconcile(ctx context.Context, cluster *eto
...
+		logger.Error(err, "Failed to reconcile the Suite Runner role binding")
+		return err
+	}
 	return err
 }
 
-// reconcileSuiteRunnerServiceAccount will reconcile the ETOS SuiteStarter service account to its expected state.
-func (r *ETOSSuiteStarterDeployment) reconcileSuiteRunnerServiceAccount(ctx context.Context, logger logr.Logger, name types.NamespacedName, owner metav1.Object) (*corev1.ServiceAccount, error) {
-	target := r.serviceaccount(name)
+// reconcileSuiteRunnerRole will reconcile the ETOS SuiteStarter role to its expected state.
```
Comment (t-persson) [inline internal/etos/suitestarter/suitestarter.go:141]:
```suggestion
// reconcileSuiteRunnerRole will reconcile the ETOS suite runner role to its expected state.
```

File: internal/etos/suitestarter/suitestarter.go [hunk 4]
```diff
@@ -56,26 +56,31 @@ func NewETOSSuiteStarterDeployment(spec etosv1alpha1.ETOSSuiteStarter, scheme *r
...
-	logger := log.FromContext(ctx, "Reconciler", "ETOSSuiteStarter", "BaseName", name)
-	namespacedName := types.NamespacedName{Name: name, Namespace: cluster.Namespace}
-	_, err = r.reconcileSuiteRunnerServiceAccount(ctx, logger, namespacedName, cluster)
+	suiteStarterName := fmt.Sprintf("%s-etos-suite-starter", cluster.Name)
+	suiteRunnerName := fmt.Sprintf("%s-etos-suite-runner", cluster.Name)
+
+	logger := log.FromContext(ctx, "Reconciler", "ETOSSuiteStarter", "BaseName", suiteStarterName)
+
+	nsSuiteStarterName := types.NamespacedName{Name: suiteStarterName, Namespace: cluster.Namespace}
+	nsSuiteRunnerName := types.NamespacedName{Name: suiteRunnerName, Namespace: cluster.Namespace}
```
Comment (t-persson) [inline internal/etos/suitestarter/suitestarter.go:65]:
```suggestion
	suiteStarterName := types.NamespacedName{Name: fmt.Sprintf("%s-etos-suite-starter", cluster.Name), Namespace: cluster.Namespace}
	suiteRunnerName := types.NamespacedName{Name: fmt.Sprintf("%s-etos-suite-runner", cluster.Name), Namespace: cluster.Namespace}

	logger := log.FromContext(ctx, "Reconciler", "ETOSSuiteStarter", "BaseName", suiteStarterName.Name)
```

File: internal/etos/suitestarter/suitestarter.go (hunk 1 above)
Comment (andmat900) [reply internal/etos/suitestarter/suitestarter.go]: logr.Logger does not have .Infof()

Comment (t-persson) [reply internal/etos/suitestarter/suitestarter.go:344]: Ah right. Sorry. Should be:

```suggestion
		logger.Info("Creating a new role binding for the suite starter", "roleName", roleName)
```

**Review by t-persson — APPROVED** [review]

**Pull request merged**